### DISCUSSION TITLE
About addGeneIntegrationMatrix() RNA cell assignments

### DISCUSSION THREAD

**Discussion by tzhu-bio** [GreenleafLab/ArchR#1040] [General]
Why the predictedGroup after prediction with RNA is not the same as the Object in RNA. 

For example, cellA is **Epithelial cell** in RNAObject, but cellA is **T cell** in predictedGroup.

**Comment by rcorces** [comment]
I do not understand the question. If your ATAC-seq and RNA-seq data are acquired from different cells, then cellA does not exist in both objects

**Reply by tzhu-bio** [reply]
Sorry, I didn't describe the question clearly.

After addGeneIntegrationMatrix, ArchR will assign a RNA cell to each ATAC cell. **predicteGroup** represents the celltype in RNA and **predictedCell** is the cell name in RNA. I found that in RNA, cellA corresponds to T cell, but in the **predictedCell**  cellA corresponds to Epithelial cell in the **predicteGroup**.

**Reply by rcorces** [reply]
In theory, this should not be possible. If you can create a reproducible example using the tutorial data, I'd be happy to take a look.

**Reply by abs51295** [reply]
@rcorces I am also facing a similar issue. I have multi-omic data and I analyzed RNA and ATAC separately. The overlap after processing both the modalities is ~50% and hence I use the integration method to label the remaining cells. As you can see in the image, the x axis is the predicted group and the y axis is the original label for the matched cells. So, what's celltype A in RNA gets celltype B in ATAC. 

One possible reason: because these are all T-cell subclusters, the lineage between T cell subtypes is pretty hard to predict?

Another possible reason: I wasn't able to run the multiome workflow because of a known bug that's gonna be fixed in the next release. Hence, I relied on the GeneScoreMatrix computed by archR for integration. 

Thanks

![image](https://user-images.githubusercontent.com/13438222/198144420-3abf8952-9ac9-4456-aac9-a202f5d6b891.png)

**Reply by rcorces** [reply]
@abs51295 

> One possible reason: because these are all T-cell subclusters, the lineage between T cell subtypes is pretty hard to predict?

Yes - I dont think GeneScores would be accurate enough for this

> Another possible reason: I wasn't able to run the multiome workflow because of a known bug that's gonna be fixed in the next release.

I'm not sure exactly what you're referring to but I would encourage you to use the `dev` branch rather than wait for the next final release if you need that bug fix.
